src/codec.rs

```diff
@@ -1223,17 +1261,99 @@ macro_rules! impl_codec_through_iterator {
 	)*}
 }
 
-impl_codec_through_iterator! {
+// Constants from rust's source:
+// https://doc.rust-lang.org/src/alloc/collections/btree/node.rs.html#43-45
+const BTREE_B: usize = 6;
+const BTREE_CAPACITY: usize = 2 * BTREE_B - 1;
+const BTREE_MIN_LEN_AFTER_SPLIT: usize = BTREE_B - 1;
+
+/// Estimate the mem size of a btree.
+fn mem_size_of_btree<T>(len: u32) -> usize {
```

Comment:
Would be nice to have some sort of test for this, but also not that easy to implement. We would require some custom allocator to get the statistics.

Reply:
But not a requirement for this pr

Reply:
Good point ! Added a simple test and also adjusted `mem_size_of_btree()` a bit.